server/src/stateManagement.mjs

```diff
@@ -179,38 +178,38 @@ function getUserId(userId){
   return userId;
 }
 
-// return state based on hierarchy (From lowest priority to highest) global->group->user
-function getState(userId) {
+// return state based on hierarchy (From lowest priority to highest) global->group->user if mergedState=true
+//return state of the UserId if mergedState=false,to update the state
+function getState(userId,mergedState = true) {
   userId = "" + userId;
-
-  //to get the userId from given user/appId
+//to get the userId from given user/appId
   userId = getUserId(userId);
-  if ( userId in state ) {
-    const stateCopy = JSON.parse( JSON.stringify(state) )
-    let finalState = stateCopy['global'];
-    userId = '' + userId;
-    if( userId.includes("~")){
-      let group = "~"+userId.split("~")[1];
-      if (group.includes("#")){
-        group = group.split("#")[0];
+  if(!mergedState){
+    return state[''+userId];
+  }else{
+    if ( userId in state ) {
+      const stateCopy = JSON.parse( JSON.stringify(state) )
+      let finalState = stateCopy['global'];
+      userId = '' + userId;
+      if( userId.includes("~")){
```

Comment:
Can we please encapsulate all the "parsing" of userIds into coreUserId, group, appId into one utility function (or set of utility functions) so that code that's doing higher order things don't have to be cluttered by that kind of lower-level parsing logic?

Reply:
@michaelcfine,I have used the existing parse(userId) in userManagment.mjs,to parse the userId and get appId, group,then assigned those values to the finalstate object.